examples/device-allocator.cpp

```diff
@@ -4,29 +4,66 @@
 //
 // SPDX-License-Identifier: (MIT)
 //////////////////////////////////////////////////////////////////////////////
-#include "umpire/DeviceAllocator.hpp"
+#include "umpire/util/device_allocator_helper.hpp"
 #include "umpire/ResourceManager.hpp"
 
-__global__ void my_kernel(umpire::DeviceAllocator alloc, double** data_ptr)
+/*
+ * Very simple kernel that uses only the first thread to "get" the
+ * existing DeviceAllocator and to allocate an array of ints.
+ * Making sure that the data_ptr is pointing to the device allocated array,
+ * it sets one element to a value that will be checked later.
+ */
+__global__ void my_kernel(double** data_ptr)
 {
   if (threadIdx.x == 0) {
+    umpire::DeviceAllocator alloc = umpire::util::getDeviceAllocator(0);
     double* data = static_cast<double*>(alloc.allocate(10 * sizeof(double)));
     *data_ptr = data;
-
     data[7] = 1024;
   }
 }
 
+__global__ void my_other_kernel(double** data_ptr)
+{
+  if (threadIdx.x == 0) {
+    umpire::DeviceAllocator alloc = umpire::util::getDeviceAllocator("my_device_alloc");
+    double* data = static_cast<double*>(alloc.allocate(1 * sizeof(double)));
+    *data_ptr = data;
+    data[0] = 42; 
+  }
+}
+
 int main(int argc, char const* argv[])
 {
   auto& rm = umpire::ResourceManager::getInstance();
+
+  //Checking to make sure a DeviceAllocator doesn't yet exist
+  if(!umpire::util::existsDeviceAllocator()) {
+    std::cout << "Before I create a DeviceAllocator, it doesn't exist!" << std::endl;
```

Comment:
Is this an error or is this the expected case?  I'm not sure what the message means.  It appears that this is information-only, but I am not sure.  If it is information-only, then perhaps an else-block below with a different output might also be useful.